components/mintmaker/production/base/manager_patches.yaml

```diff
@@ -5,6 +5,15 @@ metadata:
   namespace: system
 spec:
   template:
+    metadata:
+      annotations:
+        # We store the renovate image reference in this annotation to leverage Kustomize's
+        # built-in "images" transformer. Kustomize's "images" feature works natively with
+        # image references (e.g., container images in spec.containers[].image), but it does
+        # not support updating env variables directly, unless we write a new transformer
+        # plugin. By storing the image reference here, we can dynamically update it with
+        # Kustomize, and the container can access it using 'env[].valueFrom.fieldRef'.
+        mintmaker-renovate-image: quay.io/konflux-ci/mintmaker-renovate-image:latest
```

Comment:
I feel like this is a little confusing that we patch the controller pod to add this annotation, then we use the image transformer to replace this image reference, I'll put this in controller's repo instead of patching it here.